This patch reverts 597333f1024092b94bcd8772541e19a0f707bd40 (http://www.sqlite.org/src/info/597333f102)

Rely on HAVE_POSIX_FALLOCATE value determined by configure.
posix_fallocate() won't become available just because
(_XOPEN_SOURCE >= 600 || _POSIX_C_SOURCE >= 200112L)
evaluates to true.

--- sqlite3.c
+++ sqlite3.c
@@ -22931,13 +22931,6 @@
 */
 #if SQLITE_OS_UNIX              /* This file is used on unix only */
 
-/* Use posix_fallocate() if it is available
-*/
-#if !defined(HAVE_POSIX_FALLOCATE) \
-      && (_XOPEN_SOURCE >= 600 || _POSIX_C_SOURCE >= 200112L)
-# define HAVE_POSIX_FALLOCATE 1
-#endif
-
 /*
 ** There are various methods for file locking used for concurrency
 ** control:
